include/deal.II/matrix_free/tensor_product_kernels.h

```diff
@@ -3526,28 +3536,31 @@ namespace internal
       }
     else if (dim == 1)
       {
-        return (1. - p[0]) * values[0] + p[0] * values[1];
+        return Number3(values[0]) + p[0] * Number3(values[1] - values[0]);
```

Comment:
Is the different cast strategy for values only and values and gradients intentional? If yes can you explain why?

Reply:
Good point, I forgot about it.